app/models/user.rb

```diff
@@ -1536,8 +1529,14 @@ def number_of_deleted_posts
   end
 
   def number_of_flagged_posts
-    ReviewableFlaggedPost.where(target_created_by: self.id).count
+    posts
+      .with_deleted
```

Comment:
Discovered during more exploratory testing that the default scopes on Post result in the previous `flags_received_count` not actually counting flags received on deleted posts. I've opted to keep the original `number_of_flagged_posts` behaviour of counting those, since the signal is centred around flags moderators deem as _accurate_.